python/lancedb/embeddings.py

```diff
@@ -16,15 +16,22 @@
 from typing import Callable, Union
 
 import numpy as np
-import pandas as pd
 import pyarrow as pa
 from lance.vector import vec_to_table
 from retry import retry
 
+from .util import safe_import_pandas
+
+pd = safe_import_pandas()
+if pd is not None:
+    DATA = Union[pa.Table, pd.DataFrame]
+else:
+    DATA = pa.Table
```

Comment:
Would this not work fine? Or would it failing type hint checks on downstream apps if they didn't have Pandas installed?
```suggestion
DATA = Union[pa.Table, "pd.DataFrame"]
```